### DISCUSSION TITLE
NextAuth Pages newUser conditional redirection.

### DISCUSSION THREAD

**Discussion by ePaTo22** [nextauthjs/next-auth#11741] [Help]
I'm using the Pages configuration from next-auth (https://next-auth.js.org/configuration/pages) to redirect new users to a specific path on their first sign-in. However, I want to apply this redirection only if the user originally comes from the '/' path. If the user arrives from a different path, like profile/settings, I want to redirect them to that specific path instead of the default one. How can I do this properly?

**Comment by BertramYe** [comment]
don't use the `try...catch....` to wrap the function of the `signIn` , and then use the `redirect` function of the callbacks to do this , or you can use the `authorized` of the callbacks to do this , and combined with the middileware
